lib/server-infra/headers.js

```diff
@@ -0,0 +1,10 @@
+"use strict";
+
+module.exports = async (ctx, next) => {
+  // These headers should be kept (mostly) in sync with configuration and tests
+  // in https://github.com/whatwg/misc-server.
+  ctx.set("Strict-Transport-Security", "max-age=63072000; includeSubDomains; preload");
+  ctx.set("X-Content-Type-Options", "nosniff");
+  ctx.set("X-Frame-Options", "deny");
```

Comment:
Do we want to avoid setting this on non-HTML responses, where it has no impact? You could check `if (ctx.type.startsWith("text/html"))`

Reply:
It will have impact, no? Otherwise those responses could be framed.

Reply:
For the misc-server configs we send it unconditionally. If it's merely a no-op for some kinds of responses, isn't that fine?

Reply:
True, I guess I meant, where it's not needed.

Reply:
Old somewhat-similar discussion seems to have concluded in favor of using headers more often than not: https://github.com/whatwg/misc-server/issues/90